### DISCUSSION TITLE
Representing regularly spaced coordinates?

### DISCUSSION THREAD

**Discussion by anntzer** [pydata/xarray#8550] [Q&A] [answered]
Say I have data measured on a regularly spaced grid, e.g.
```python
array = DataArray(np.zeros((10, 10)), {"x": np.arange(10) * .1, "y": np.arange(10) * .1})
```
One issue with this representation is that it is slightly awkward to retrieve back the grid spacings dx and dy.  Of course one can just do `array.x[1] - array.x[0]`, but this can run into (standard) issues with floating point errors:
```python
# take a subarray
sub = da[:-2, 2:]
print((sub.x[1] - sub.x[0]) - (sub.y[1] - sub.y[0]))
```
returns `<xarray.DataArray ()> array(-2.77555756e-17)` so one gets forced to add some slop if one wants e.g. to assert that the grid is indeed square.

Is there some way to represent the regularly spaced grid simply as a (start, stop, step) triplet (similarly to how stdlib python's range objects are represented?)

(An alternative approach could be to just use x=0...(nx-1), y=0...(ny-1), and put dx, dy into array.attrs, but that doesn't play well with slicing e.g. `array[::2, ::2]` which doesn't update the attributes.)

Looking forward to your suggestions.

**Comment by keewis** [comment, marked as answer]
see #8473 for a recent discussion on this topic

**Reply by anntzer** [reply]
Ah, great, thanks!  I searched for arange but not for linspace in the issue tracker :)  I guess I'll just track that issue (which seems basically equivalent to what I'm looking for), then.
